Update private contest UI and fix creation flow of contest and supabase api

Chitransh1011 opened the pull request:
# Title
Enhancements to contest modes, creation flow, and UI integration with Supabase for a more reliable and consistent contest experience.
Issue #20 

# Changes

## Private Contests

- Added optional fields (problem count, rating range, contest type).
- Ensured created_at is always available for proper date display.
- Applied default values for missing fields.

## Supabase Updates

- Added new columns to support extended contest metadata.
- Updated queries to return all fields required by the UI.
- Filtered contests to show only public or group-related contests.
- Converted date fields to ISO strings to fix “Invalid Date” errors.

## UI Fixes

- Improved display for dates, rating, problem count, and participants.
- Added safe handling for missing/null data to prevent UI breakage.
- Refined private contest card layout for consistency.

##Contest Creation

- Validates required fields (name, start date/time, rating range).
- Dynamically calculates end time based on contest type.
- Safely handles optional fields.
- Provides clear success and error feedback.
## Impact
- Extended database schema with better contest metadata.
- More stable and user-friendly contest creation and display.
- Consistent and reliable private contest handling.

<img width="1141" height="678" alt="Screenshot 2025-10-03 115832" src="https://github.com/user-attachments/assets/973777f3-2191-4b55-949f-1488c11d1bd0" />

chitransh2023ug1011-coder commented:
Hey @Hackeries I have fixed some issue of backend [Error: You cannot use different slug names for the same dynamic path ('pathId' !== 'sectionId').] Now It is working fine and contest is createing with saving in database also but should I commit with different branch or same branch

Chitransh1011 commented:
Hey @Hackeries now I have fixed the conflict as you have requested but still getting error can you describe the issue and I face some issue while syncing the fork that [Error: You cannot use different slug names for the same dynamic path ('pathId' !== 'sectionId').] and fixed this error and now contest is easily creating and saving in supabase db

Chitransh1011 commented:
<img width="1829" height="852" alt="image" src="https://github.com/user-attachments/assets/9efa469e-9eda-429b-a09c-9cb467cd369d" />
you can see the screenshot also

Chitransh1011 commented:
Hey @Hackeries no response from you what should I do I have sync fork it and again pr it .

Chitransh1011 commented:
Hey @Hackeries now review it is any conflict in pr or not and if not  review my code . I have ensured to first sync fork and git pull it and then update anything in the code

Hackeries reviewed (changes requested):
<img width="1309" height="327" alt="image" src="https://github.com/user-attachments/assets/93541bbb-fd9f-4619-9c02-e5a8701b741f" />


<img width="371" height="131" alt="image" src="https://github.com/user-attachments/assets/9eb89dbf-532b-4b94-8f37-0628f030118b" />


<img width="319" height="105" alt="image" src="https://github.com/user-attachments/assets/031797cb-7c1d-4d7a-9764-e017a433c5b6" />



There is a deployment error bevause of that i cannot preview..
please fix this and make uh also add the schema also..
because in the current schema the private contest is not working..correctly..

do one thing Fork the repo again instead of doing sync fork fork the whole repo again..

and then update the fixes..

I'll merge your PR then..

Chitransh1011 commented:
Hey @Hackeries  does you have made create_contest.sql correctly because now issue of rls is coming
